README.md

```diff
@@ -52,6 +52,20 @@ if err != nil {
 ```
 Full example: [samples/middleware.go](samples/middleware.go)
 
+### Proof of Possession
+Additionally, you may want to make sure, whether you have been called by a trust-worthy consumer. Trust between applications/services is established with certificates in principle. So, in case of mTls based communication, you can check, whether the token was issued for the consumer. This can be done by
+ performing the JWT Certificate Thumbprint X5t confirmation method's validation. See specification [RFC 8705](https://tools.ietf.org/html/rfc8705#section-3.1). It can be done in the following manner:
+ 
+```go
+func myEndpoint(w http.ResponseWriter, r *http.Request) {
+    err := auth.ValidateX5tThumbprint(auth.ClientCertificateFromCtx(r), auth.TokenFromCtx(r))
```

Comment:
For a feature version (like 1.0) we can think of solution how to integrate this check into the middleware.

In our project we check all authentication with the help of http subrouters and middlewares before the actual endpoint.

```go
subrouter.Use(
    middleware.AuthenticationHandler
)
```

A way to customise this Middleware and enable/disable parts of the validation based on the route would be really cool. Maybe something like the following:

```go
func (m *Middleware) CreateAuthenticationHandler(proofOfPossession options) func(handler http.Handler) http.Handler {
    // return the actual AuthenticationHandler with options enabled/disabled
    // and checks like azp == expected caller client id
}
```

Reply:
see issue #38